airsonic-main/src/main/java/org/airsonic/player/service/SecurityService.java

```diff
@@ -336,15 +384,114 @@ protected boolean isFileInFolder(String file, String folder) {
         return file.toUpperCase().startsWith(folder.toUpperCase());
     }
 
-    public void setSettingsService(SettingsService settingsService) {
-        this.settingsService = settingsService;
+
+    // =======================================================================================================
+    // Utilities for Sonos link.
+
+    public Authentication authenticate(String username, String password){
+        UsernamePasswordAuthenticationToken authReq = new UsernamePasswordAuthenticationToken(username, password);
+        return authenticationManager.authenticate(authReq);
     }
 
-    public void setUserDao(UserDao userDao) {
-        this.userDao = userDao;
+    /**
+     * Generate a link code, and put in the cache for future use.
+     *
+     * @param householdId from sonos, represent a user on sonos
+
+     * @return The link code.
+     */
+    public String generateLinkCode(String householdId){
+        String linkCode = createLinkCode();
+
+        sonosLinkcodeCache.put(new Element(linkCode, householdId));
+
+        return linkCode;
     }
 
-    public void setUserCache(Ehcache userCache) {
-        this.userCache = userCache;
+    public String getHousehold(String linkCode){
+        Element element = sonosLinkcodeCache.get(linkCode);
+
+        if(element == null){
+            return  null;
+        } else {
```

Comment:
nit: No need for an `if…{return}else{return}` construct if you're returning, an `if…{return} return` construct is enough :)

Reply:
It's a habit, equality is easier to understand than a negation.

I change it as your way.